### PR TITLE
PLAT-3006 Add unbound policy test to opentdf/tests

### PR DESCRIPTION
**Pull request by mustyantsev**

### CODE REVIEW

**Review by sievdokymov-virtru — COMMENTED** [review]
@mustyantsev think need to add this test into CI test scope

File: xtest/sdk/py/test_unbound_policy.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            for folder_name, subfolders, filenames in os.walk(tdf_nopolicy_folder):
+                for filename in filenames:
+                    file_path = os.path.join(folder_name, filename)
+                    zip_ref.write(file_path, os.path.relpath(file_path, tdf_nopolicy_folder))
+
+
+        sampleTdfStorage = TDFStorageType()
+        sampleTdfStorage.set_tdf_storage_file_type(os.path.join(cwd, updated_tdf_file))
+        client.decrypt_file(sampleTdfStorage, os.path.join(cwd, "sample_policy.txt"))
+
```
Comment (dmihalcik-virtru) [inline xtest/sdk/py/test_unbound_policy.py:78]:
```suggestion
        client.decrypt_file(sampleTdfStorage, os.path.join(cwd, "sample_policy.txt"))
        log.error("Broken policy ignored")
        sys.exit(1)
```

File: xtest/sdk/py/test_unbound_policy.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+def update_policy_binding(zip_file_path, new_policy_binding):
+    with open(zip_file_path, "r") as manifest:
+            manifest_data = json.load(manifest)
+            manifest_data['encryptionInformation']['keyAccess'][0]['policyBinding'] = new_policy_binding
+
+    with open(zip_file_path, "w") as manifest:
+            json.dump(manifest_data, manifest)
+
+def main():
+    try:
```
Comment (dmihalcik-virtru) [inline xtest/sdk/py/test_unbound_policy.py:24]: Move this `try` down to the decrypt_file line below

File: xtest/sdk/py/test_unbound_policy.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+                    file_path = os.path.join(folder_name, filename)
+                    zip_ref.write(file_path, os.path.relpath(file_path, tdf_nopolicy_folder))
+
+
+        sampleTdfStorage = TDFStorageType()
+        sampleTdfStorage.set_tdf_storage_file_type(os.path.join(cwd, updated_tdf_file))
+        client.decrypt_file(sampleTdfStorage, os.path.join(cwd, "sample_policy.txt"))
+
+    except Exception as e:
+        error_message = f"An error occurred: {e}"
```
Comment (dmihalcik-virtru) [inline xtest/sdk/py/test_unbound_policy.py:80]: Add a validation here that this exception has an expected message. I don't know what that might be, and raise if not

File: xtest/sdk/py/test_unbound_policy.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    # Get the policy binding from the second zip file
+        fullpath = os.path.join(tdf_with_policy_folder, '0.manifest.json')
+        policy_binding = extract_policy_binding(fullpath)
+
+    # Update the policy binding in the first zip file
+        fullpath = os.path.join(tdf_nopolicy_folder, '0.manifest.json')
+        update_policy_binding(fullpath, policy_binding)
+
+    # Zip the updated files
+        updated_tdf_file = "updated_file.tdf"
```
Comment (dmihalcik-virtru) [inline xtest/sdk/py/test_unbound_policy.py:67]: You might want to add some more logging to support better debugging of the CI log files

**Review by dmihalcik-virtru — CHANGES_REQUESTED** [review]

File: xtest/sdk/py/test_unbound_policy.py [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+            for filename in filenames:
+                file_path = os.path.join(folder_name, filename)
+                zip_ref.write(file_path, os.path.relpath(file_path, tdf_nopolicy_folder))
+                logging.info("Added file to zip: %s", file_path)
+
+
+    sampleTdfStorage = TDFStorageType()
+    sampleTdfStorage.set_tdf_storage_file_type(os.path.join(cwd, updated_tdf_file))
+    try:
+        client.decrypt_file(sampleTdfStorage, os.path.join(cwd, "sample_policy.txt"))
```
Comment (dmihalcik-virtru) [inline xtest/sdk/py/test_unbound_policy.py:84]:
```suggestion
        client.decrypt_file(sampleTdfStorage, os.path.join(cwd, "sample_policy.txt"))
        log.error("Broken policy ignored")
        sys.exit(1)
```

**Review by dmihalcik-virtru — CHANGES_REQUESTED** [review]

**Review by patmantru — DISMISSED** [review]

File: xtest/sdk/py/test_unbound_policy.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+        logging.error("Broken policy ignored")
+        sys.exit(0)
```
Comment (dmihalcik-virtru) [inline xtest/sdk/py/test_unbound_policy.py:86]:
```suggestion
        assert False, "Broken policy should raise exception on attempt to decrypt"
```

You'll need some kind of check to see that the file with a broken policy was *not* decrypted, e.g. make sure it is not present in the output. You can raise an error here, for example, or check the output file does not exist

**Review by dmihalcik-virtru — CHANGES_REQUESTED** [review]

**Review by dmihalcik-virtru — APPROVED** [review]

File: xtest/sdk/py/test_unbound_policy.py [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+        os.remove(os.path.join(cwd, "sample_policy.txt"))
+
+    try:
+        client.decrypt_file(sampleTdfStorage, os.path.join(cwd, "sample_policy.txt"))
+        if os.path.exists(os.path.join(cwd, "sample_policy.txt")):
+            logging.error("Decryption was successful, but it was not expected due to a broken policy.")
+            sys.exit(1)
+        else:
+            logging.error("Broken policy ignored")
+            sys.exit(0)
```
Comment (dmihalcik-virtru) [inline xtest/sdk/py/test_unbound_policy.py:94]: To me this indicates an error in the SDK. It should raise an exception. When does this happen?

**Review by dmihalcik-virtru — APPROVED** [review]

**Pull request merged**